nautobot/core/views/mixins.py

```diff
@@ -110,6 +118,20 @@ def check_permissions(self, request):
         if not self.has_permission():
             self.handle_no_permission()
 
+    def check_if_queryset_attribute_exist(self):
+        # Check if self.queryset is specified in the ModelViewSet before performing subsequent actions
+        # If not, display an error message
+        try:
+            self.get_queryset()
+        except AssertionError:
+            messages.error(self.request, "Please include a `queryset` attribute in '%s'" % self.__class__.__name__)
+
+    def check_if_table_class_attribute_exist(self):
+        # Check if self.table_class is specified in the ModelViewSet before performing subsequent actions
+        # If not, display an error message
+        if getattr(self, "table_class") is None:
+            messages.error(self.request, "Please include a `table_class` attribute in '%s'" % self.__class__.__name__)
```

Comment:
Should either of these set `self.has_error`? Or in general how should issues with these two cases get handled by the view class? Like I see that `form_valid` calls `self.check_if_queryset_attribute_exist()`, but then it goes right ahead with doing a `self.queryset` call afterwards, which will still raise an unhandled exception to the UI?

Reply:
I think it might be a good idea to have a separate template for missing attributes. Because the UI would not render in the first place if `self.queryset` is set, so `messages.error()` is not shown.